ros_ws/src/autonomous/include/wall_following.h

```diff
@@ -4,95 +4,68 @@
 #include <iostream>
 
 #include "drive_msgs/drive_param.h"
-#include "drive_msgs/pid_input.h"
+#include "pid_controller.h"
 #include "sensor_msgs/LaserScan.h"
 #include "std_msgs/Bool.h"
 #include "std_msgs/Float64.h"
 #include <ros/console.h>
 #include <ros/ros.h>
 
+constexpr const char* TOPIC_DRIVE_PARAMETERS = "/input/drive_param/wallfollowing";
+constexpr const char* TOPIC_LASER_SCAN = "/scan";
+constexpr const char* TOPIC_EMERGENCY_STOP = "/emergency_stop";
+
+constexpr int LIDAR_SAMPLE_COUNT = 720;
+
+// Discard lidar measurements outside this range
+constexpr float MIN_RANGE = 0.2;
+constexpr float MAX_RANGE = 30;
+
+constexpr float FALLBACK_RANGE = 4;
```

Comment:
This is not fail-safe. If the range a undefined lidar-index is requested, the car should not drive on without noticing that something went wrong. It would be safer to set this value to 0.
See also emergency_stop.cpp line 16: rangeAtDegree(). Maybe there should also be used this const?
```suggestion
constexpr float FALLBACK_RANGE = 0;
```

Reply:
I agree that the fallback value is problematic. Setting it to zero is not good either, since the point of the min and max values it that you can expect the number to be between them. 0 will lead to a division by zero in the wallfollowing code.

Anyway, the point of this pullrequest is to make more readable what happens in the wallfollowing code without changing its behaviour. I will soon make a second pulllrequest where fallbacks aren't used.